Tools/Scripts/webkitdirs.pm

```diff
@@ -917,6 +917,8 @@ sub availableXcodeSDKs
 
 sub isValidXcodeSDKPlatformName($) {
     my $name = shift;
+    # strip sdk version and variant qualifiers prior to validating platform
+    $name =~ s/(\d+\.[\d\.]+)?(\.internal)?$//;
```

Comment:
I don't think stripping the version in the validation function makes sense, because then we'd have to make sure that the rest of `webkitdirs` can handle when $xcodeSDKPlatformName contains a version number. What I would recommend instead is a modification to `determineXcodeSDKPlatformName`:

```diff
     if (checkForArgumentAndRemoveFromARGVGettingValue("--sdk", \$sdk)) {
         $xcodeSDK = lc $sdk;
         $xcodeSDKPlatformName ||= $sdk;
-        $xcodeSDKPlatformName =~ s/\.internal$//;
+        $xcodeSDKPlatformName =~ s/(\d+\.[\d\.]+)?(\.internal)?$//;
         die "Couldn't determine platform name from Xcode SDK" unless  isValidXcodeSDKPlatformName($xcodeSDKPlatformName);
         return;
     }
```

Reply:
that seems more correct to me assuming that's what gets passed to the validation code, I will try to verify in a bit